designer/client/src/common/ProcessUtils.ts

```diff
@@ -244,15 +267,10 @@ class ProcessUtils {
         }
     };
 
-    findNodeDefinitionIdOrType = (node: NodeType) => this.findNodeDefinitionId(node) || node.type || null;
-
-    getNodeBaseTypeCamelCase = (node: NodeType) => node.type && node.type.charAt(0).toLowerCase() + node.type.slice(1);
-
     findNodeConfigName = (node: NodeType): string => {
         // First we try to find id of node (config for specific custom node by id).
-        // If it is falsy then we try to extract config name from node type (config for build-in components e.g. variable, join).
-        // If all above are falsy then it means that node is special process properties node without id and type.
-        return this.findNodeDefinitionId(node) || this.getNodeBaseTypeCamelCase(node) || "$properties";
+        // When is missing it means that node is special process properties node without id
```

Comment:
```suggestion
        // When it is missing it means that the node is a special process properties node without id
```
and the old comment above should be 
` // First we try to find the id of the node (configs for specific custom nodes are resolved by id).
`

articles are hard

Reply:
Yep, this comment had more problems than only articles, I rephrased it more deeply.